src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * @file src/Modules/Settings/Uexport.php
+ */
+
+namespace Friendica\Module\Settings;
+
+use Friendica\App;
+use Friendica\App\Arguments;
+use Friendica\BaseModule;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Module\BaseSettingsModule;
+
+/**
+ * Module to export user data
+ **/
+class Uexport extends BaseSettingsModule
+{
+	/**
+	 * Handle the request to export data.
+	 * At the moment one can export two different data set
+	 * 1. The profile data that can be used by uimport to resettle
+	 *    to a different Friendica instance
+	 * 2. The entire data-set, profile plus postings
+	 *
+	 * If there is an action required through the URL / path, react
+	 * accordingly and export the requested data.
+	 **/
+	public static function content()
+	{
+		parent::content();
+		$args = self::getClass(Arguments::class);
+		if ($args->getArgc() == 3) {
+			// @TODO Replace with router-provided arguments
+			$action = $args->get(2);
+			header("Content-type: application/json");
+			header('Content-Disposition: attachment; filename="' . $a->user['nickname'] . '.' . $action . '"');
```

Comment:
`$a` isn't defined at this point.

Reply:
But it works as expected...

Reply:
What is the displayed name of the file the browser makes you download? No notice shows in the PHP error log?

Reply:
I've downloaded the `friendica1.account` file. But I'll see where to get the username elsewhere from so that this is no issue.

Reply:
It's available as `self::getApp()->user`